src/superagent-promise-plugin.js

```diff
@@ -3,8 +3,7 @@ function endPromise(req) {
 
   return new _Promise(function (resolve, reject) {
     req.end(function (err, res) {
-      err = err || res.error;
-      if (err) reject(err);
+      if (err || res.error) reject(res);
```

Comment:
Sounds good, but though the res could be needed even when err is defined. 
For example, I'm returning an error code 422 when a validation error occurs in my application, and this fulfills the `err` parameter with 'Error : Unprocessable entity'. But I do need the res body in order to know the specifics.

The err seems to be only a text describing the error, am I right ? This can be found in the res.statusText.

I don't know if there is a way to prevent breaking change ?